hazelcast-client/src/test/java/com/hazelcast/client/proxy/DistributedObjectListenerTest.java

```diff
@@ -79,4 +82,19 @@ public void run() throws Exception {
         }, 5);
     }
 
+
+    @Test
+    public void testGetDistributedObjectsAfterRemove() {
+        HazelcastInstance server = hazelcastFactory.newHazelcastInstance();
+        IMap firstMap = server.getMap("firstMap");
+        server.getMap("secondMap");
+
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient();
+        assertEquals(2, client.getDistributedObjects().size());
+
+        firstMap.destroy();
```

Comment:
we can assume this has the same affect as a second client destroying the proxy but since the reported error was generated by using the clients I would recommend adding that test here as well. i.e. just get a second client verify it has 2 objects initially, then let other client destroy a proxy and verify that this client sees this removal as expected.